IPython/core/interactiveshell.py

```diff
@@ -2781,6 +2784,15 @@ def run_ast_nodes(self, nodelist, cell_name, interactivity='last_expr',
         if not nodelist:
             return
 
+        if interactivity == 'last_expr_or_assign':
+            if isinstance(nodelist[-1], ast.Assign):
+                asg = nodelist[-1]
+                if len(asg.targets) == 1:
+                    nnode = ast.Expr(ast.Name(asg.targets[0].id, ast.Load()))
```

Comment:
I think this will crash if you try to do a multiple assignment.  When I parse something like `(a, b) = (1, 2)`, I get an assignment with a length-1 list of targets whose only entry is an instance of `ast.Tuple`, which has no `id`:

```
In [5]: sys.version_info
Out[5]: sys.version_info(major=3, minor=5, micro=2, releaselevel='final', serial=0)

In [6]: %%ast
   ...: (a, b) = (1, 2)
   ...:
Module(
  body=[
    Assign(
      targets=[
        Tuple(
          elts=[
            Name(id='a', ctx=Store()),
            Name(id='b', ctx=Store()),
          ],
          ctx=Store(),
        ),
      ],
      value=Tuple(
        elts=[
          Num(1),
          Num(2),
        ],
        ctx=Load(),
      ),
    ),
  ],
)